libcaf_test/caf/test/runner.hpp

```diff
@@ -26,8 +30,10 @@ class CAF_TEST_EXPORT runner {
 
 private:
   parse_cli_result parse_cli(int argc, char** argv);
+  std::optional<std::regex> parse_regex_opt(const std::string& regex_string);
```

Comment:
This isn't accessing any state, so it can be `static`. With this name, I would expect that the argument is actually the name of the option, as was my suggestion. Passing in the option value is also ok, but then this should have a different name, e.g., `to_regex`.